src/Categories/Category/Monoidal/Instance/Endofunctors.agda

```diff
@@ -0,0 +1,131 @@
+open import Data.Unit.Polymorphic using (⊤)
+open import Function using (case_of_; flip)
+open import Level using (Lift; lift; _⊔_)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl)
+
+open import Categories.Morphism using (_≅_; Iso)
+open import Categories.Category.Cartesian using (Cartesian)
+open import Categories.Category.Cartesian.Monoidal using (module CartesianMonoidal)
+open import Categories.Category.Core using (Category)
+open import Categories.Category.Cocartesian using (Cocartesian)
+open import Categories.Category.Construction.Functors using (Functors)
+open import Categories.Functor using (Functor; _∘F_) renaming (id to idF)
+open import Categories.NaturalTransformation using (NaturalTransformation; _∘ₕ_)
+open import Categories.Category.Monoidal using (Monoidal; monoidalHelper)
+open import Categories.Category.Monoidal.Symmetric using (Symmetric; symmetricHelper)
+open import Categories.NaturalTransformation.NaturalIsomorphism using (niHelper)
+
+open import Categories.NaturalTransformation.NaturalIsomorphism
+  using (NaturalIsomorphism; unitorˡ; unitorʳ; associator)
+
+import Categories.Morphism.Reasoning as MR
+
+Endofunctors : Category o ℓ e → Category (o ⊔ ℓ ⊔ e) (o ⊔ ℓ ⊔ e) (o ⊔ e)
+Endofunctors C = Functors C C
+
+module _ where
+  -- Conversion function from natural isomorphisms to isomorphisms in the functor category.
+  ≃→≅ : ∀ {C D : Category o ℓ e}
+      → {F G : Functor C D}
+      → NaturalIsomorphism F G
+      → Categories.Morphism._≅_ (Functors C D) F G
+  ≃→≅ i = record
+    { from = i.F⇒G
+    ; to = i.F⇐G
+    ; iso = record { isoˡ = λ {x} → Iso.isoˡ (i.iso x) ; isoʳ = λ {x} → Iso.isoʳ (i.iso x) }
+    } where module i = NaturalIsomorphism i
+
+module _ where
+  Endofunctors-Monoidal : ∀ {C : Category o ℓ e} → Monoidal (Endofunctors C)
+  Endofunctors-Monoidal {C} = record
+    { ⊗ = record
+      { F₀ = uncurry′ _∘F_
+      ; F₁ = uncurry′ _∘ₕ_
+      ; identity = λ { {F , G} →
+        let module F = Functor F
+            module G = Functor G in
+          begin F.F₁ id ∘ id ≈⟨ F.identity ⟩∘⟨refl ⟩
+                id ∘ id ≈⟨ identity² ⟩
+                id ∎ }
+      ; homomorphism = λ { {F , G} {F₂ , G₂} {F₃ , G₃} {f₁ , f₂} {g₁ , g₂} →
+        let module F₃ = Functor F₃
+            module F₂ = Functor F₂
+            module G = Functor G
+            module G₂ = Functor G₂
+            module f₁ = NaturalTransformation f₁
+            module f₂ = NaturalTransformation f₂
+            module g₁ = NaturalTransformation g₁
+            module g₂ = NaturalTransformation g₂ in
+          begin
+            F₃.₁ (g₂.η _ ∘ f₂.η _) ∘ g₁.η (G.₀ _) ∘ f₁.η (G.₀ _) ≈⟨ pushˡ F₃.homomorphism ⟩
+            F₃.₁ (g₂.η _) ∘ F₃.₁ (f₂.η _) ∘ g₁.η (G.₀ _) ∘ f₁.η (G.₀ _) ≈⟨ refl⟩∘⟨ extendʳ (g₁.sym-commute _) ⟩
+            F₃.₁ (g₂.η _) ∘ g₁.η (G₂.F₀ _) ∘ F₂.F₁ (f₂.η _) ∘ f₁.η (G.₀ _) ≈⟨ sym-assoc ⟩
+            (F₃.₁ (g₂.η _) ∘ g₁.η (G₂.₀ _)) ∘ F₂.₁ (f₂.η _) ∘ f₁.η (G.₀ _) ∎ }
```

Comment:
I think the general rule is to factor out large inline-definitions like these into explicit, stand-alone definitions, either in a `where` block after the record definition, or in a `private` block before `Endofunctors-Monoidal`. If the definition is useful in itself, it doesn't even need to be private.

The same goes for the definitions below.